### PR TITLE
CombineImports option

### PR DESCRIPTION
**Pull request by anhany**
When upgrading to Webpack3, I would get weird circular dependency errors. Combining the internal imports to reference the index file fixed this.

### CODE REVIEW

**Comment by anhany** [comment]
FYI: Waiting to publish the nuget after a code review.

File: Program.cs [hunk 1]
```diff
@@ -20,14 +20,15 @@ class Program
...
-      Typescripter.exe <SOURCE> <DESTINATION> [<APIPATH> [ --httpclient ]]
+      Typescripter.exe <SOURCE> <DESTINATION> [<APIPATH> [ --httpclient ] [ --combineimports ]]
                        [--files=<FILES> | --class=<CLASSNAMES>]...
       Typescripter.exe ( -h | --help )
 
     Options:
       --files=<FILES>         Comma seperated list of .dll files to generate models from. [ default: *.client.dll ]
       --class=<CLASSNAMES>    Comma seperated list of controller class names. [ default: ApiController ]
       --httpclient            Generated data service will use the new HttpClientModule for angular 4.
+	  --combineimports        Combines model imports to come from the generated index, rather than individual model files. [default: false]
```
Comment (danw8) [inline Program.cs]: white space matters here. Does this have the correct spacing?

File: Generators/DataServiceGenerator.cs [hunk 2]
```diff
@@ -34,12 +34,20 @@ public static List<string> Generate(string apiRelativePath, List<Type> apiContro
...
+				sb.AppendLine("import {");
+				foreach (var import in models.OrderBy(m => m.Name)) {
+					sb.AppendLine(string.Format("\t{0},", import.Name));
+				}
+				sb.AppendLine("} from './';\n");
+			} else {
+				foreach (var m in models.OrderBy(m => m.Name)) {
+					sb.AppendLine("import { " + m.Name + " } from '.';");
+				}
+				sb.AppendLine("");
```
Comment (danw8) [inline Generators/DataServiceGenerator.cs]: This got moved inside the loop. Is that what we want?

File: Generators/EntityGenerator.cs [hunk 3]
```diff
@@ -34,10 +34,20 @@ public static class EntityGenerator {
...
+				sb.AppendLine("import {");
+				foreach(var import in importTypes) {
+					sb.AppendLine(string.Format("\t{0},", import));
+				}
+				sb.AppendLine("} from './';\n");
+			} else {
+				foreach (var import in importTypes) {
+					sb.AppendLine(string.Format("import {{ {0} }} from './{0}';", import));
+				}
+				sb.AppendLine();
```
Comment (danw8) [inline Generators/EntityGenerator.cs:49]: This got moved inside the loop. Is that what we want?

**Pull request merged**